src/views/Checkout.vue

```diff
@@ -217,44 +215,57 @@ export default class Checkout extends Vue {
             accountInfo = walletInfo.accounts.get(contractInfo.owner.toUserFriendlyAddress());
         }
 
-        // FIXME: Currently unused, but should be reactivated
-        this.$store.commit('setActiveAccount', {
+        // FIXME: Also handle active account we get from store
+        this.$setActiveAccount({
             walletId: walletInfo.id,
             userFriendlyAddress: accountInfo!.userFriendlyAddress,
         });
 
-        // The next block is the earliest for which tx are accepted by standard miners
-        const validityStartHeight = this.height + 1
-            - TX_VALIDITY_WINDOW
-            + this.request.validityDuration;
-
-        const request: KeyguardClient.SignTransactionRequest = {
-            layout: 'checkout',
-            shopOrigin: this.rpcState.origin,
-            appName: this.request.appName,
-            shopLogoUrl: this.request.shopLogoUrl,
-
-            keyId: walletInfo.keyId,
-            keyPath: accountInfo!.path,
-            keyLabel: walletInfo.label,
-
-            sender: (contractInfo || accountInfo!).address.serialize(),
-            senderType: contractInfo ? contractInfo.type : Nimiq.Account.Type.BASIC,
-            senderLabel: (contractInfo || accountInfo!).label,
-            recipient: this.request.recipient.serialize(),
-            recipientType: this.request.recipientType,
-            // recipientLabel: '', // Checkout is using the shopOrigin instead
-            value: this.request.value,
-            fee: this.request.fee,
-            validityStartHeight,
-            data: this.request.data,
-            flags: this.request.flags,
-        };
-
-        staticStore.keyguardRequest = request;
+        this.proceedToTransactionSigning(walletInfo, accountInfo!, contractInfo);
+    }
 
-        const client = this.$rpc.createKeyguardClient();
-        client.signTransaction(request);
+    private proceedToTransactionSigning(walletInfo: WalletInfo, accountInfo: AccountInfo, contractInfo?: ContractInfo) {
```

Comment:
I removed this redundant function splitting in #214 as well: https://github.com/nimiq/accounts/pull/214/files#diff-10870cb1266d47895e2ffe5e572207e0L189

Reply:
I reintroduced again as I liked it now that the method now has become longer.

Reply:
As the `setAccountOrContract` method got shorter after some refactoring, thus I merged the methods again.

Reply:
As the `setAccountOrContract` method got shorter after some refactoring, thus I merged the methods again.